readthedocs/templates/builds/build_detail.html

```diff
@@ -227,7 +227,13 @@ <h3>{% trans "Build Errors" %}</h3>
         <h3>{% trans "Error" %}</h3>
         <p class="build-error"
            data-bind="text: error">
-          {{ build.error }}
+            {#
+              I'd like to use ``build.error|urlize`` here, so we can have nice links.
+              However, this is not possible because we are using `data-bind="text: error"`
+              which means that Knockout.js will use the `.error` attribute to fill
```

Comment:
What error attribute? On the build itself?

Reply:
Yes

Reply:
If you would like this, we should make the APIv3 response match what we want, perhaps applying the filter on the field in the response there.

Reply:
Hrm.. I don't think the API should return blob HTML tags on its response.

Reply:
Similar to templates and HTMX, the front end should only have to display the data from the backend application. I don't think we should have a front end reimplementation `urlize`, which is required here to show `build.error` in the same way the templates would render it.

On the implementation side, I think we should be authoring error message to be a bit more rich than what we're doing now. Relying on `urlize` results in error message HTML like:

`Something happened. See our documentation on webhooks at <a href="https://docs.readthedocs.io/page/something.html">https://docs.readthedocs.io/page/something.html</a>`

This is not super user friendly, but is also not accessible either. Using HTML in the error would be better, but does require some translation to HTML at some point:

`Something happened. <a href="https://docs.readthedocs.io/page/something.html">Read our documentation on webhooks</a> for more information`

This is perhaps another good point for the notification redesign.

Reply:
> Similar to templates and HTMX, the front end should only have to display the data from the backend application

I think this is out of discussion. We are not using HTMX which requires specific endpoints for its purpose. Besides, here we are talking about a _resource API_ that exposes data to our users. If we want to start exposing "data to be rendered by the frontend without any modification" that's a whole different use case than what the API was thought for.

In the case we want move in that direction,we will need to add `_display` fields (or similar) to all the things we want to expose to user as-is.

Reply:
I know we're not using HTMX, but we've talked about this approach specifically. Templates are more applicable then, and maintain a similar opinionated approach. My point is that we should do what we can at the backend in order to keep our frontend footprint low.

*I think we talk about this more as part of the notification refactor either way. I just brought this up here to note that replicating `urlize` in front end code is not a great path forward.*

> If we want to start exposing "data to be rendered by the frontend without any modification"

I don't feel I'm talking about the same thing here. I am only referring to text with HTML links embedded, not any more structure than that.

> Besides, here we are talking about a resource API that exposes data to our users

Well, I'd certainly say our API is dual function. The API needs to be useful for our own application too, it's not just a user resource API.

> that's a whole different use case than what the API was thought for

Perhaps, but our application is also a consumer of this API. We don't need all the features exposed to end users though.

But if not the current API, it should be something on the application side. This can be separate, a different API or API-like endpoint, but it should be part of the application. I guess I don't see a strong reason for avoiding APIv3 though.

> In the case we want move in that direction,we will need to add _display fields (or similar)

That seems fine, I don't have strong opinions on structure. We're not solving this problem here, so it's worth thinking about more. I'm just noting here as I think we should change our patterns, and the pattern noted in the PR here, in our notification refactor.